packages/clerk-js/src/ui/contexts/ClerkUIComponentsContext.tsx

```diff
@@ -91,7 +91,9 @@ export const useSignUpContext = (): SignUpContextType => {
     displayConfig: displayConfig,
   });
 
-  let signUpUrl = pickRedirectionProp('signUpUrl', { ctx, options, displayConfig }, false);
+  let signUpUrl =
+    (ctx.routing === 'path' ? ctx.path : undefined) ||
+    pickRedirectionProp('signUpUrl', { options, displayConfig }, false);
```

Comment:
Removed the `ctx` from passing it inside the `pickRedirectionProp` here, as it never contains a `signUpUrl` prop from the `SignUp` component. The priority now is:

- The `ctx.path` is the `path` prop that user passes to the `SignUp` component, when using `routing=path`.
- The `options` passed in `pickRedirectionProp` is the ClerkProvider `signUpUrl` prop or the `NEXT_PUBLIC_CLERK_SIGN_UP_URL` env var, with that order.
- Lastly, the `displayConfig` that has the Account Portal value.

cc @nikosdouvlis @dimkl

Reply:
I'm a bit skeptical because the logic here has become somewhat complicated.
I think the reason you're not moving the path check inside `pickRedirectionProp` is that `pickRedirectionProp` is too generic  for that?